### PR TITLE
Make the request/response limits configurable

### PR DESCRIPTION
**Pull request by andersjanmyr**
```
	fsthttp.Limits.SetMaxHeaderValueLen(32 * 1024)
```

### CODE REVIEW

File: fsthttp/limits.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import "github.com/fastly/compute-sdk-go/internal/abi/fastly"
+
+// Limits handles HTTP limits
+var Limits = limits{}
+
+type limits struct {
+}
+
+// GetMaxHeaderNameLen gets the header name limit
+func (limits) GetMaxHeaderNameLen() int {
```
Comment (dgryski) [inline fsthttp/limits.go:12]: Effective Go recommends not having getter function names start with `Get`.  This should jus be `MaxHeaderNameLen()`.

**Review by dgryski — CHANGES_REQUESTED** [review]
A small comment on function naming, but otherwise this looks great.  Thanks!

File: fsthttp/limits.go (hunk 1 above)
Comment (andersjanmyr) [reply fsthttp/limits.go]: Yes, I agree, it reads better. :)

File: fsthttp/limits.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package fsthttp
+
+import "github.com/fastly/compute-sdk-go/internal/abi/fastly"
+
+// Limits handles HTTP limits
+var Limits = limits{}
+
+type limits struct {
+}
```
Comment (dkegel-fastly) [inline fsthttp/limits.go:9]: I'd put the limits in the struct, that's four fewer globals...

Comment (andersjanmyr) [reply fsthttp/limits.go:9]: It would create a circular dependency between the `fsthttp` and `internal/abi/fastly` packages.

Comment (dkegel-fastly) [reply fsthttp/limits.go:9]: emoji-homer-hedge :-)

**Review by dgryski — APPROVED** [review]
LGTM

**Comment by dgryski** [comment]
I'm going to see if I can tie the limits to a particular Request object.  That would solve having large allocations for every request (if it's not needed), and also prevent any data races from modifying these global variables from different goroutines.

**Pull request merged**